worlds/dkc2/Locations.py

```diff
@@ -0,0 +1,707 @@
+    "Mudhole Marsh": {location for location in all_locations.keys() if "Mudhole Marsh" in location},
+    "Rambi Rumble": {location for location in all_locations.keys() if "Rambi Rumble" in location},
+    "King Zing Sting": {location for location in all_locations.keys() if "King Zing Sting" in location},
+    "Haunted Hall": {location for location in all_locations.keys() if "Haunted Hall" in location},
+    "Gusty Glade": {location for location in all_locations.keys() if "Gusty Glade" in location},
+    "Parrot Chute Panic": {location for location in all_locations.keys() if "Parrot Chute Panic" in location},
+    "Web Woods": {location for location in all_locations.keys() if "Web Woods" in location},
+    "Kreepy Krow": {location for location in all_locations.keys() if "Kreepy Krow" in location},
+    "Arctic Abyss": {location for location in all_locations.keys() if "Arctic Abyss" in location},
+    "Windy Well": {location for location in all_locations.keys() if "Windy Well" in location},
+    "Castle Crush": {location for location in all_locations.keys() if "Castle Crush" in location},
+    "Clapper's Cavern": {location for location in all_locations.keys() if "Clapper's Cavern" in location},
+    "Chain Link Chamber": {location for location in all_locations.keys() if "Chain Link Chamber" in location},
+    "Toxic Tower": {location for location in all_locations.keys() if "Toxic Tower" in location},
+    "Stronghold Showdown": {location for location in all_locations.keys() if "Stronghold Showdown" in location},
+    "Screech's Sprint": {location for location in all_locations.keys() if "Screech's Sprint" in location},
+    "Jungle Jinx": {location for location in all_locations.keys() if "Jungle Jinx" in location},
+    "Black Ice Battle": {location for location in all_locations.keys() if "Black Ice Battle" in location},
+    "Klobber Karnage": {location for location in all_locations.keys() if "Klobber Karnage" in location},
+    "Fiery Furnace": {location for location in all_locations.keys() if "Fiery Furnace" in location},
+    "Animal Antics": {location for location in all_locations.keys() if "Animal Antics" in location},
+    "Swanky": {location for location in all_locations.keys() if "Swanky" in location},
+}
+
+def setup_locations(world: "DKC2World"):
+    location_table = {
+        **stage_clears,
+        **stage_bonus,
+    }
+
+    if world.options.dk_coin_checks:
+        location_table.update(stage_dk_coin)
+    if world.options.kong_checks:
+        location_table.update(stage_kong)
+    if world.options.swanky_checks:
+        location_table.update(swanky_quiz)
+    if world.options.balloonsanity:
+        location_table.update(balloonsanity)
+    if world.options.coinsanity:
+        location_table.update(coinsanity)
+    if world.options.bananasanity:
+        location_table.update(bananasanity)
+
+    return location_table
+
+lookup_id_to_name: typing.Dict[int, str] = {id: name for name, _ in all_locations.items()}
```

Comment:
```suggestion
```
This comprehension doesn't define `id`, so its value is the builtin function with that name, meaning that this map only has one (not very useful) value in it. This is apparently copied from some of @PoryGone's worlds which also have it currently. It doesn't seem to actually get used, so I think it could just be removed (as well as in `Items.py` although that one's not broken).